### PR TITLE
refactor: streamlines exit exception handling

### PR DESCRIPTION
**Pull request by jpower432**
## Description

- Create a custom exception for input validation issues in entrypoint classes
- Updates entrypoint class `run` functions with try/except blocks with exception/exit handling using the `handle_exception` function.
- Add check for existence of trestle_root directory (separate commit)

Stacked/Incorporates #98 in response to feedback. 

## Type of change

- [X] Refactor/Maintenance

## How has this been tested?

- [X] Unit tests for autosync updated to reflect proper exit codes and error messages.
- [X] E2E (just to check for regressions because the entrypoint files are being changed)
    - https://github.com/jpower432/trestle-bot/actions/runs/7186479037
    - https://github.com/jpower432/cautious-potato/actions/runs/7186522609/job/19572095854 (`with GitHub Actions`)

## Checklist

- [X] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jpower432** [comment]
@JimFuller-RedHat PTAL

File: trestlebot/entrypoints/autosync.py [hunk 1]
```diff
@@ -95,63 +100,71 @@ def validate_args(self, args: argparse.Namespace) -> None:
...
-        )
-
-        # Assuming an edit has occurred assemble would be run before regenerate.
-        # Adding this to the list first
-        if not args.skip_assemble:
-            assemble_task = AssembleTask(
-                authored_object=authored_object,
-                markdown_dir=args.markdown_path,
-                model_filter=model_filter,
+        exit_code: int = SUCCESS_EXIT_CODE
```
Comment (JimFuller-RedHat) [inline trestlebot/entrypoints/autosync.py:125]: Minor nit: is there any reason not to directly use SUCCESS_EXIT_CODE ?

File: trestlebot/entrypoints/autosync.py [hunk 2]
```diff
@@ -95,63 +100,71 @@ def validate_args(self, args: argparse.Namespace) -> None:
...
+                    markdown_dir=args.markdown_path,
+                    model_filter=model_filter,
+                )
+                pre_tasks.append(regenerate_task)
+            else:
+                logger.info("Regeneration task skipped.")
+
+            super().run_base(args, pre_tasks)
+        except Exception as e:
+            exit_code = handle_exception(e)
```
Comment (JimFuller-RedHat) [inline trestlebot/entrypoints/autosync.py:165]: ignore my last comment ;)

**Review by JimFuller-RedHat — APPROVED** [review]
**LGTM**

**Review by beatrizmcouto — APPROVED** [review]
Reviewed during paired code review Dec 13

**Pull request merged**